package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,503 @@
+        self._ag2 = mda.AtomGroup([ag[1] for ag in self._torsions])
+        self._ag3 = mda.AtomGroup([ag[2] for ag in self._torsions])
+        self._ag4 = mda.AtomGroup([ag[3] for ag in self._torsions])
+
+        if bat_nc is not None:
+            self.load_bat(bat_nc)
+
+    def _prepare(self):
+        self.bat = []
+
+    def _single_frame(self):
+        # Calculate coordinates based on the root atoms
+        # The rotation axis is a normalized vector pointing from atom 0 to 1
+        # It is described in two degrees of freedom
+        # by the polar angle and azimuth
+        if (self._root.dimensions[:3] == 0).any():
+            (p0, p1, p2) = self._root.positions
+        else:
+            (p0, p1, p2) = make_whole(self._root, inplace=False)
+        v01 = p1 - p0
+        v21 = p1 - p2
+        # Internal coordinates
+        r01 = np.sqrt(np.sum(v01 *
+                             v01))  # Distance between first two root atoms
+        r12 = np.sqrt(np.sum(v21 *
+                             v21))  # Distance between second two root atoms
+        a012 = np.arccos(max(-1.,min(1.,np.sum(v01*v21)/\
+          np.sqrt(np.sum(v01*v01)*np.sum(v21*v21))))) # Angle between root atoms
+        # Exernal coordinates
+        e = v01 / r01
+        phi = np.arctan2(e[1], e[0])  # Polar angle
+        theta = np.arccos(e[2])  # Azimuthal angle
+        # Rotation to the z axis
+        cp = np.cos(phi)
+        sp = np.sin(phi)
+        ct = np.cos(theta)
+        st = np.sin(theta)
+        Rz = np.array([[cp * ct, ct * sp, -st], [-sp, cp, 0],
+                       [cp * st, sp * st, ct]])
+        pos2 = Rz.dot(p2 - p1)
+        # Angle about the rotation axis
+        omega = np.arctan2(pos2[1], pos2[0])
+        root_based = np.concatenate((p0, [phi, theta, omega, r01, r12, a012]))
+
+        # Calculate internal coordinates from the torsion list
+        bonds = calc_bonds(self._ag1.positions,
+                           self._ag2.positions,
+                           box=self._ag1.dimensions)
+        angles = calc_angles(self._ag1.positions,
+                             self._ag2.positions,
+                             self._ag3.positions,
+                             box=self._ag1.dimensions)
+        torsions = calc_dihedrals(self._ag1.positions,
+                                  self._ag2.positions,
+                                  self._ag3.positions,
+                                  self._ag4.positions,
+                                  box=self._ag1.dimensions)
+        # When appropriate, calculate improper torsions
+        shift = torsions[self._primary_torsion_indices]
+        shift[self._unique_primary_torsion_indices] = 0.
+        torsions -= shift
+        # Wrap torsions to between -np.pi and np.pi
+        torsions = ((torsions + np.pi) % (2 * np.pi)) - np.pi
+
+        self.bat.append(np.concatenate((root_based, bonds, angles, torsions)))
+
```

Comment:
How about adding a `_conclude` method to cast `bat` to an array? You're doing it already for `save` and `load` and it seems a much more sensible thing, especially in order to slice it along time:
```suggestion
    def _conclude(self):
        self.bat = np.asarray(self.bat, dtype=np.float64)
```

On that note, I would make the collection list `self._bat` to make sure that users understand that they don't need it (and also safer for parallel code) and then have
```python

    def _conclude(self):
        self.bat = np.asarray(self._bat, dtype=np.float64)
        self._bat = []
```

Reply:
Now I'm initiating it as an array, not a list.